config/initializers/apipie.rb

```diff
@@ -1,178 +1,180 @@
-
-  config.translate = lambda do |str, loc|
-    old_loc = FastGettext.locale
-    FastGettext.set_locale(loc)
-    trans = _(str) if str
-    FastGettext.set_locale(old_loc)
-    trans
-  end
-  config.help_layout = 'apipie_dsl/apipie_dsls/help.html.erb'
-end
-
-Apipie.configure do |config|
-  config.app_name = "Foreman"
-  config.app_info = "The Foreman is aimed to be a single address for all machines life cycle management."
-  config.copyright = ""
-  config.api_base_url = "/api"
-  config.api_controllers_matcher = ["#{Rails.root}/app/controllers/api/**/*.rb"]
-  config.ignored = []
-  config.ignored_by_recorder = %w[]
-  config.doc_base_url = "/apidoc"
-  config.use_cache = Rails.env.production? || File.directory?(config.cache_dir)
-  # config.languages = [] # turn off localized API docs and CLI, useful for development
-  config.languages = ENV['FOREMAN_APIPIE_LANGS'].try(:split, ' ') || FastGettext.available_locales
-  config.default_locale = FastGettext.default_locale
-  config.locale = ->(loc) { loc ? FastGettext.set_locale(loc) : FastGettext.locale }
-
-  substitutions = {
-    :operatingsystem_families => Operatingsystem.families.join(", "),
-    :providers => -> { ComputeResource.providers.keys.join(', ') },
-    :providers_requiring_url => -> { ComputeResource.providers_requiring_url },
-    :default_nic_type => InterfaceTypeMapper::DEFAULT_TYPE.humanized_name.downcase,
-    :template_kinds => -> { Rails.cache.fetch("template_kind_names", expires_in: 1.hour) { TemplateKind.pluck(:name).join(", ") } },
-    :host_rebuild_steps => -> { Host::Managed.valid_rebuild_only_values.join(', ') },
-  }
-
-  config.translate = lambda do |str, loc|
-    old_loc = FastGettext.locale
-    FastGettext.set_locale(loc)
-    if str
-      trans = _(str)
-      trans = trans % Hash[substitutions.map { |k, v| [k, v.respond_to?(:call) ? v.call : v] }]
-    end
-    FastGettext.set_locale(old_loc)
-    trans
-  end
-  config.validate = false
-  config.force_dsl = true
-  config.reload_controllers = Rails.env.development?
-  config.markup = Apipie::Markup::Markdown.new if Rails.env.development? && defined? Maruku
-  config.default_version = "v2"
-  config.update_checksum = true
-  config.checksum_path = ['/api/', '/apidoc/']
-end
-
-# check apipie cache in dev mode
-if Foreman.in_rake?('apipie:cache', 'apipie:cache:index', 'config')
-  # No need to check the cache if we're regenerating or handling config
-elsif Apipie.configuration.use_cache
-  cache_name = File.join(Apipie.configuration.cache_dir, Apipie.configuration.doc_base_url + '.json')
-  if File.exist?(cache_name)
-    target = File.mtime(cache_name)
-    roots = ::Rails::Engine.subclasses.map { |e| e.instance.root }
-    roots << Rails.root
-    outdated = roots.any? do |root|
-      path = "#{root}/app/controllers/api"
-      File.exist?(path) && Find.find(path).any? do |e|
-        File.mtime(e) > target
+Rails.application.config.to_prepare do
```

Comment:
I believe `Apipie` config lives over the code reload, so we should use after_initialize here, otherwise we uselessly reconfigure the `Apipie`
Also IIUC only the substitutions have to wait after_initilize, so we might even do two configure blocks and only the second with `config.translate = ` would go in `after_initialize` block.

All the other code should be fine out of the block if I read correctly.